doc/release-notes.md

```diff
@@ -74,6 +74,15 @@ P2P and network changes
   node using P2P relay. This version reduces the initial broadcast guarantees
   for wallet transactions submitted via P2P to a node running the wallet. (#18038)
 
+- The Tor onion service that is automatically created (`-listenonion`) will now
+  be created as a Tor v3 service instead of Tor v2. The private key that was used
+  for Tor v2 (if any) will be left untouched in `onion_private_key` in `-datadir`
+  and can be removed if not needed. Bitcoin Core will no longer attempt to read it.
+  The private key for the Tor v3 service will be saved in `onion_v3_private_key`.
+  To use the deprecated Tor v2 service (not recommended), then `onion_private_key`
+  can be copied over `onion_v3_private_key`, e.g.
+  `cp -f onion_private_key onion_v3_private_key`.
```

Comment:
1f06a2901d1ee924b95f482e8693842ee07863de Perhaps inform users of the debug log messages, e.g. `ERROR: DeserializeDB: Deserialize or I/O error` and `Invalid or missing peers.dat`, that they may see if they use Tor v2 henceforth.

Reply:
Added a release notes mention.